code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceAddRemoveLinkServiceValidationFilter.java

```diff
@@ -0,0 +1,59 @@
+package io.cattle.platform.servicediscovery.api.filter;
+
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.api.dao.ServiceConsumeMapDao;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+public class ServiceAddRemoveLinkServiceValidationFilter extends AbstractDefaultResourceManagerFilter {
+
+    @Inject
+    ServiceConsumeMapDao consumeMapDao;
+
+    private static final Map<String, Boolean> ACTIONS;
+
+    static {
+        ACTIONS = new HashMap<>();
```

Comment:
@sonchang as the datastructure is used only within the same class, and the field you refer to can have just 2 values, I think enum would be an overkill